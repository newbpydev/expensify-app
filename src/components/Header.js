import React from "react";
import { BrowserRouter, Routes, Route, Link, NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expensify</h1>

    <nav>
      <ul>
        <li>
          <NavLink
            className={({ isActive }) => (isActive ? "is-active" : "")}
            to="/"
          >
            Dashboard
          </NavLink>
        </li>
        <li>
          <NavLink
            className={({ isActive }) => (isActive ? "is-active" : "")}
            to="create"
          >
            Create Expense
          </NavLink>
        </li>
        {/* <li>
          <NavLink
            className={({ isActive }) => (isActive ? "is-active" : "")}
            to="edit"
          >
            Edit Expense
          </NavLink>
        </li> */}
        <li>
          <NavLink
            className={({ isActive }) => (isActive ? "is-active" : "")}
            to="help"
          >
            Help
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);


export default Header;