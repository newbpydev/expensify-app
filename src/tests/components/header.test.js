import React from "react";
// import ReactShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";
// import toJson from "enzyme-to-json";
import Header from "../../components/Header";

test("should render header correctly", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();

  // expect(wrapper.find("h1").text()).toBe("Expensify")
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);

  // expect(renderer.getRenderOutput()).toMatchSnapshot();

  // console.log(renderer.getRenderOutput())
});
