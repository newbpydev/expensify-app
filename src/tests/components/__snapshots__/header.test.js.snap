// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <ul>
      <li>
        <NavLink
          className={[Function]}
          to="/"
        >
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink
          className={[Function]}
          to="create"
        >
          Create Expense
        </NavLink>
      </li>
      <li>
        <NavLink
          className={[Function]}
          to="help"
        >
          Help
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
